x-pack/plugins/ml/public/application/explorer/anomaly_timeline.tsx

```diff
@@ -0,0 +1,380 @@
+          },
+          onDragStart(e) {
+            let target = e.target as HTMLElement;
+            while (target && target !== document.body && !target.classList.contains('sl-cell')) {
+              target = target.parentNode as HTMLElement;
+            }
+            if (ALLOW_CELL_RANGE_SELECTION && target !== document.body) {
+              dragSelect$.next({
+                action: DRAG_SELECT_ACTION.DRAG_START,
+              });
+              disableDragSelectOnMouseLeave.current = false;
+            }
+          },
+          onElementSelect() {
+            if (ALLOW_CELL_RANGE_SELECTION) {
+              dragSelect$.next({
+                action: DRAG_SELECT_ACTION.ELEMENT_SELECT,
+              });
+            }
+          },
+        }),
+      []
+    );
+
+    const {
+      filterActive,
+      filteredFields,
+      maskAll,
+      overallSwimlaneData,
+      selectedCells,
+      viewByLoadedForTimeFormatted,
+      viewBySwimlaneData,
+      viewBySwimlaneDataLoading,
+      viewBySwimlaneFieldName,
+      viewBySwimlaneOptions,
+      swimlaneLimit,
+      selectedJobs,
+    } = explorerState;
+
+    const setSwimlaneSelectActive = useCallback((active: boolean) => {
+      if (isSwimlaneSelectActive.current && !active && disableDragSelectOnMouseLeave.current) {
+        dragSelect.stop();
+        isSwimlaneSelectActive.current = active;
+        return;
+      }
+      if (!isSwimlaneSelectActive.current && active) {
+        dragSelect.start();
+        dragSelect.clearSelection();
+        dragSelect.setSelectables(document.querySelectorAll('.sl-cell'));
+        isSwimlaneSelectActive.current = active;
+      }
+    }, []);
+    const onSwimlaneEnterHandler = () => setSwimlaneSelectActive(true);
+    const onSwimlaneLeaveHandler = () => setSwimlaneSelectActive(false);
+
+    // Listens to render updates of the swimlanes to update dragSelect
+    const swimlaneRenderDoneListener = useCallback(() => {
+      dragSelect.clearSelection();
+      dragSelect.setSelectables(document.querySelectorAll('.sl-cell'));
+    }, []);
+
+    // Listener for click events in the swimlane to load corresponding anomaly data.
+    const swimlaneCellClick = useCallback((selectedCellsUpdate: any) => {
+      // If selectedCells is an empty object we clear any existing selection,
+      // otherwise we save the new selection in AppState and update the Explorer.
+      if (Object.keys(selectedCellsUpdate).length === 0) {
+        setSelectedCells();
+      } else {
+        setSelectedCells(selectedCellsUpdate);
+      }
+    }, []);
+
+    const showOverallSwimlane =
+      overallSwimlaneData !== null &&
+      overallSwimlaneData.laneLabels &&
+      overallSwimlaneData.laneLabels.length > 0;
+
+    const showViewBySwimlane =
+      viewBySwimlaneData !== null &&
+      viewBySwimlaneData.laneLabels &&
+      viewBySwimlaneData.laneLabels.length > 0;
+
+    return (
+      <>
+        <EuiPanel paddingSize="s">
```

Comment:
I think you should add a panel around the 'Anomalies' section too (charts and table) as otherwise the swimlane section is given too much prominence for me. @mdefazio any thoughts?